### PR TITLE
Added decimalItemType validator to validator and modified booleanItemType validator to support "1" and "0" as boolean types.

### PR DESCRIPTION
**Pull request by reepoi**

### CODE REVIEW

File: oblib/validator.py [hunk 1]
```diff
@@ -77,12 +77,29 @@ def xbrli_boolean_item_type_validator(value, concept):
     """XBRLI boolean validator."""
     errors = []
     if type(value) is str:
-        if value.lower() not in ['true', 'false']:
+        if re.search(r'\d', value):
```
Comment (cwhanse) [inline oblib/validator.py]: Revert the changes to `xbrli_boolean_item_type_validator`, we're adding this capability in @michael-bbs 's branch.

File: oblib/validator.py [hunk 2]
```diff
@@ -77,12 +77,29 @@ def xbrli_boolean_item_type_validator(value, concept):
...
     elif type(value) is not bool:
         errors += ["'{}' is not a valid boolean value.".format(value)]
     return errors
 
+def xbrli_decimal_item_type_validator(value, concept):
+    """XBRLI decimal validator"""
+    errors = []
+    if type(value) is str:
+        if re.search(r'[^0-9.]', value):
+            errors += ["'{} is not a valid decimal value.".format(value)]
```
Comment (cwhanse) [inline oblib/validator.py]: Use `errors.append("'{} is not a valid decimal value.".format(value))` instead of `+= [...]`

File: oblib/validator.py [hunk 3]
```diff
@@ -77,12 +77,29 @@ def xbrli_boolean_item_type_validator(value, concept):
...
+        elif len(re.findall(r'(\d*\.\d+|\d+)', value)) is not 1:
+            errors += ["'{} is not a valid decimal value.".format(value)]
+    elif type(value).__name__ not in ['float', 'int']:
+        errors += ["'{} is not a valid decimal value.".format(value)]
+    return errors
```
Comment (cwhanse) [inline oblib/validator.py]: Need to set a valid `value`, and return `value, errors`

File: oblib/tests/test_validator.py [hunk 4]
```diff
@@ -31,13 +31,32 @@ def test_validate_concept_value(self):
 
         concept.type_name = "xbrli:booleanItemType"
         self.assertEqual(1, len(validator.validate_concept_value(concept, "Arf")))
+        self.assertEqual(1, len(validator.validate_concept_value(concept, "0100")))
```
Comment (cwhanse) [inline oblib/tests/test_validator.py]: We're changing the signature of each validator function to return a tuple `value, errors`, so all test statements need to change.  I'd suggest this pattern:

result = validator.validate_concept_value(concept, test_value)
self.assertEqual(expected, result[0])
self.assertTrue(result[1])  or self.assertFalse(result[1]) if the test is expected to fail

File: oblib/validator.py [hunk 5]
```diff
@@ -77,12 +77,29 @@ def xbrli_boolean_item_type_validator(value, concept):
...
+        elif value.lower() not in ['true', 'false']:
+            errors += ["'{}' is not a valid boolean value.".format(value)]
+    elif type(value) is int:
+        if not (value is 1 or value is 0):
             errors += ["'{}' is not a valid boolean value.".format(value)]
     elif type(value) is not bool:
         errors += ["'{}' is not a valid boolean value.".format(value)]
     return errors
 
+def xbrli_decimal_item_type_validator(value, concept):
```
Comment (cwhanse) [inline oblib/validator.py]: Delete the `concept` argument, it's not being used in the low-level functions.

**Comment by jessiedeot** [comment]
@cla-bot check
